### PR TITLE
Issue/12028 post card ui

### PR DESCRIPTION
**Pull request by malinajirka**
Parent issue #12028

This PR duplicates most of the logic from [ReaderPostListAdapter.renderPost(...)](https://github.com/wordpress-mobile/WordPress-Android/blob/588b9582c3ad022b91fd1be63111ac306d76c4ef/WordPress/src/main/java/org/wordpress/android/ui/reader/adapters/ReaderPostAdapter.java#L442). I've added TODO items for the logic which hasn't been copied yet.

I've tried to keep each commit single focused so I'd recommend reviewing this PR by commits and comparing the logic against  [ReaderPostListAdapter.renderPost(...)](https://github.com/wordpress-mobile/WordPress-Android/blob/588b9582c3ad022b91fd1be63111ac306d76c4ef/WordPress/src/main/java/org/wordpress/android/ui/reader/adapters/ReaderPostAdapter.java#L442).

I realize the review of this code might not be trivial, but I couldn't think of any way how to facilitate it for the reviewer, I'm sorry. Let me know if you have any suggestions. I also believe we'll thoroughly test all kinds of posts before this code will be released to the users, so I personally wouldn't worry about it too much.

To test:
This PR doesn't touch the UI, so it can't be tested yet.

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/reader/discover/ReaderPostUiStateBuilder.kt [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+        return ReaderPostUiState(
+                id = post.postId,
+                blogUrl = buildBlogUrl(post),
+                dateLine = buildDateLine(post),
+                avatarOrBlavatarUrl = buildAvatarOrBlavatarUrl(post),
+                blogName = buildBlogName(post),
+                excerpt = buildExcerpt(post),
+                title = buildTitle(post),
+                photoFrameVisibility = buildPhotoFrameVisbility(post),
```
Comment (ashiagr) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/discover/ReaderPostUiStateBuilder.kt:43]: Np: Typo `buildPhotoFrameVisbility` => `buildPhotoFrameVisibility`

File: WordPress/src/main/java/org/wordpress/android/ui/reader/discover/ReaderPostUiStateBuilder.kt [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+                photoTitle = buildPhotoTitle(post),
+                featuredImageUrl = buildFeaturedImageUrl(post, photonWidth, photonHeight),
+                thumbnailStripUrls = buildThumbnailStripUrls(post),
+                videoOverlayVisbility = buildVideoOverlayVisbility(post),
+                // TODO malinjir Consider adding `postListType == ReaderPostListType.TAG_FOLLOWED` to showMoreMenu
+                moreMenuVisbility = accountStore.hasAccessToken(),
```
Comment (ashiagr) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/discover/ReaderPostUiStateBuilder.kt:49]: Np: Typo 
`videoOverlayVisbility` => `videoOverlayVisibility`
`moreMenuVisbility` => `moreMenuVisibility`

File: WordPress/src/main/java/org/wordpress/android/ui/reader/discover/ReaderPostUiStateBuilder.kt [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+                videoThumbnailUrl = buildVideoThumbnailUrl(post),
+                discoverSection = buildDiscoverSection(post)
+        )
+    }
+
+    private fun buildBlogUrl(post: ReaderPost) = post
+            .takeIf { it.hasBlogUrl() }
```
Comment (ashiagr) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/discover/ReaderPostUiStateBuilder.kt:56]: Nice touch, TIL 👍

File: WordPress/src/main/java/org/wordpress/android/ui/reader/discover/ReaderPostUiStateBuilder.kt (hunk 1 above)
Comment (malinajirka) [reply WordPress/src/main/java/org/wordpress/android/ui/reader/discover/ReaderPostUiStateBuilder.kt]: Fixed in 1d604acb39b501a73ee8bcfdef8a50ab4a8c634a

File: WordPress/src/main/java/org/wordpress/android/ui/reader/discover/ReaderPostUiStateBuilder.kt (hunk 2 above)
Comment (malinajirka) [reply WordPress/src/main/java/org/wordpress/android/ui/reader/discover/ReaderPostUiStateBuilder.kt]: Fixed in 1d604acb39b501a73ee8bcfdef8a50ab4a8c634a

**Comment by malinajirka** [comment]
Thanks @ashiagr ! I've fixed the typos in 1d604acb39b501a73ee8bcfdef8a50ab4a8c634a.

>While I have manually verified we're using existing conditions for elements visibility, I'll be able to test it better when we start seeing them.

💯 


It's ready for another round, thanks!

**Comment by ashiagr** [comment]
LGTM, thanks!

**Pull request merged**